modules/openthread/platform/diag.c

```diff
@@ -68,3 +75,175 @@ void otPlatDiagAlarmCallback(otInstance *aInstance)
 {
 	ARG_UNUSED(aInstance);
 }
+
+/*
+ * To enable gpio diag commands, in Devicetree create `gpio_ports` node in root path
+ * with `gpio-ports-phandles` property, which should contain phandles of GPIO port nodes.
+ * It is important to create correct Devicetree binding (more about bindings location:
+ * https://docs.zephyrproject.org/latest/build/dts/bindings.html#where-bindings-are-located),
+ * which defines `gpio-ports-phandles`
+ * type as phandles. e.g:
+ *
+ * Devicetree:
+ *
+ * gpio-ports {
+ *		compatible = "gpio-ports";
+ *		gpio-ports-phandles = <&gpio0 &gpio1>;
+ * };
+ *
+ * Devicetree binding:
+ *
+ * description: Node which contains GPIO port phandles
+ *
+ * compatible: "gpio-ports"
+ *
+ * properties:
+ *   gpio-ports-phandles:
+ *     type: phandles
+ *     required: true
```

Comment:
This does not exist AFAIK? It shouldn't be up to users to add the binding, right?

Reply:
I've created a binding yaml file and updated description.